### ISSUE TITLE
GraphQL API Attribute not removed when providing null

### ISSUE THREAD

**Issue by wbroek** [aws-amplify/amplify-js#2482]
**Describe the bug**
When doing an update mutation with a value of `null` (or undefined) for the key the attribute is not removed in DynamoDB.

**To Reproduce**
Create a model and push it and generate the graphql mutations
```
type Article @model 
{
  id: ID!
  title: String
  subtitle: String
}
```

Insert an article with a title and subtitle

Do an update mutation with a new title and set subtitle to null
```
graphqlOperation(mutations.updateArticle, {
   input: { title: 'new title', subtitle: null },
})
```

The title is changed in DynamoDB but the subtitle is still there untouched

**Expected behavior**
What was expected is that the attribute subtitle should be removed from the DynamoDB table. 
This is mentioned/documented on the [AppSync docs](https://docs.aws.amazon.com/appsync/latest/devguide/tutorial-dynamodb-resolvers.html) 
> If the attribute is set to null, it asks AWS AppSync and DynamoDB to remove that attribute from the post object

When executing the query in the AppSync console the attribute was removed (as expected)
```
mutation UpdateArticle {
  updateArticle(input: { id: "4f482b...", title: "new title", subtitle: null }) {
    id
  }
}
```

Also in the generated mutation resolver (generende with Amplify CLI) of the article there you see the line where it states when the value is null the key (attribute) should be removed
```
#if( $util.isNull($entry.value) )
    #set( $discard = $expRemove.add("#$entry.key") )
    $util.qr($expNames.put("#$entry.key", "$entry.key"))
```

**Desktop:**
 - OS: MacOS X
 - Browser Safari, Chrome, Firefox
 - Latest version

**Additional context**
Updated all amplify packages to latest with no effect, also tried to do it with the Appsync SDK and Client but no change; also with a own typed gql no change.

```
dependencies": {
    "aws-amplify": "^1.1.18",
    "aws-amplify-react": "^2.2.5",
    "aws-appsync": "^1.7.0",
    "graphql-tag": "^2.10.0",
    "node-sass": "^4.11.0",
    "npm": "^6.5.0",
    "prop-types": "^15.6.2",
    "react": "^16.7.0",
    "react-dom": "^16.7.0",
    "react-scripts": "^2.1.3"
  }
```

**Comment by elorzafe** [comment]
@wbroek 

I did the same as you and subtitle field was removed from the database.

This is my app code
```javascript
  async updateMutation() {
    const response = await API.graphql(graphqlOperation(mutations.updateArticle, {
      "input": {
        "id": "1418eeef-05d0-4fab-9f48-d9fd4e35b612",
        "title": "holanda",
        "subtitle": null
      }
    }));

    console.log({response});
  }
```

this was the response
```json
{
    "data": {
        "updateArticle": {
            "id": "1418eeef-05d0-4fab-9f48-d9fd4e35b612",
            "title": "holanda",
            "subtitle": null
        }
    }
}
```

**Comment by wbroek** [comment]
@elorzafe thank you for the check. Assuming is a local issue then; not understanding why but when creating a new react project with new amplify init it works also!

**Comment by elorzafe** [comment]
@wbroek I am glad it works

**Comment by rohit3d2003** [comment]
The same thing happens on iOS app as well

**Comment by PartyHunter** [comment]
I have the same exact issue. I am working on an android project. I must admit that I and a newbie but I am working on a simple user address field.

First I collect all the form data

>          
            addressStreet = input_user_profile_edit_address_street.text.toString()
            addressStreet2 = input_user_profile_edit_address_street2.text.toString()
            addressCity = input_user_profile_edit_address_city.text.toString()
            addressState = input_user_profile_edit_address_state.text.toString()
            addressZipcode = input_user_profile_edit_address_zipcode.text.toString()


I even ran a extra null check to ensure that the value submitted to the mutation function is null and not "" (the empty string  

> 
            addressStreet = nullCheck(addressStreet)
            addressStreet2 = nullCheck(addressStreet2)
            addressCity = nullCheck(addressCity)
            addressState = nullCheck(addressState)
            addressZipcode = nullCheck(addressZipcode)

Once all values were sanitized I ran the mutation

>             
           val updateUserInput = UpdateUserInput.builder()
          .userId(user.user_id())
          .addressStreet(addressStreet)
          .addressStreet2(addressStreet2)
          .addressCity(addressCity)
          .addressState(addressState)
           .addressZipcode(addressZipcode)
           .build()

This code works as expected. however once data has been added to the fields any attempt to insert a null value does NOT throw any errors and does not return OnFailure but also does not delete the field as expected. NOTE: When using the console submitting a null value in any field will deletes  the field as expected. This appears to be an issue with the AppSyncClient. Further more if for example the only change was to set the value of addressCity to null, the response would return as success but no change would be made and no fields would be deleted. Also if two changes were submitted in the same request, for example addressStreet was changed to a new value (21 Maple St), and addressCity was changed from "New York" to null, the response would be successful, addressStreet would be updated correctly to 21 Maple St. but addressCity would NOT be deleted and expected and would maintain its original value of "New York." 

This would seem that the code is working but the null values from the AppSyncClient is not.

I have changed my user pool mode from API Key to CognitoUserPool to see if its a fine grain permission issue but I am getting the same error. I have updated the amplify cli and even created a new AWS user profile using Amplify configure will no luck. This appears to be the only thread anywhere that is discussing this issue.
